### PR TITLE
The Bravest of Crew Get the Lustiest of Xenos

### PR DESCRIPTION
**Pull request by orthography**
## About The Pull Request

Allows dressing up the xeno queen in a maid costume (if you can get close enough [and not get torn to shreds in the 5 seconds it takes])

Possible Other Dresses (?)

- [x] Maid
- [ ] Wedding Dress
- [ ] Sundress
- [ ] Clown

From https://github.com/tgstation/tgstation/pull/18545

## Why It's Good For The Game

Adds a really goofy interaction where the weirdo crew will desperately try not to die as they dress up the xeno queen. Plus! Xeno 

image
image

## Changelog
:cl:
add: New fabrics in station clothing now allow it to fit on Xeno queens!
image: added new xeno outfits!!
fix: fixes how inspections read too.
/:cl:

### CODE REVIEW

**Comment by LukasBeedellCodestuff** [comment]
As the pig maintainer I'd like to ask, would it be possible to extend this to our current pig sprites? I think they could benefit.

**Comment by JohnFulpWillard** [comment]
> As the pig maintainer I'd like to ask, would it be possible to extend this to our current pig sprites? I think they could benefit.

We didn't have any code preventing seccies from wearing this in the first place

File: code/modules/mob/living/carbon/alien/adult/queen.dm [hunk 1]
```diff
@@ -181,3 +181,15 @@
...
+
+/mob/living/carbon/alien/adult/royal/queen/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/clothing/under/costume/maid) || istype(O, /obj/item/clothing/under/rank/civilian/janitor/maid))
+		if(do_after(user, 5 SECONDS, src))
+			maidify()
+
+/mob/living/carbon/alien/adult/royal/queen/proc/maidify()
+	name = "alien queen maid"
+	desc = "Lusty, Sexy"
+	icon_state = "alienqmaid"
```
Comment (LemonInTheDark) [inline code/modules/mob/living/carbon/alien/adult/queen.dm:193]: shouldn't this be in update_icon_state/update_name/update_desc alongside a bool on the queen?

File: code/modules/mob/living/carbon/alien/adult/queen.dm [hunk 2]
```diff
@@ -181,3 +181,15 @@
 	if(!silent)
 		to_chat(user, span_noticealien("You discard [src]."))
 	return ..()
+
+/mob/living/carbon/alien/adult/royal/queen/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/clothing/under/costume/maid) || istype(O, /obj/item/clothing/under/rank/civilian/janitor/maid))
```
Comment (LemonInTheDark) [inline code/modules/mob/living/carbon/alien/adult/queen.dm:186]: should this be a typecache

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 1 above)
Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:193]: Name updates will add the others.

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 2 above)
Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:186]: idk it works.

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 1 above)
Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:193]: sike idk how to do this

**Comment by DAKKA-WAAAGH** [comment]
SPEEDMERGE NOW!!!

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/alien/adult/queen.dm [hunk 3]
```diff
@@ -181,3 +181,17 @@
...
+		if(do_after(user, 5 SECONDS, src))
+			maidify()
+			qdel(O)
+
```
[the remark this answers is not in the document]
Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:190]: we just learned about that, you also can't even hit them

**Comment by Burger1998** [comment]
> why

because its funny

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:190]: it would make adding new cases less verbose, and overall just nicer

Comment (LemonInTheDark) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:190]: actually no scratch that bad idea I take it back

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:199]: want to talk on discord?

**Comment by Militaires** [comment]
i look at a pull request and wat the hayulllllllll, arent these the sprites i made for that one shitty pr years ago

**Comment by borb197** [comment]
will these sprites show up for xeno queens if they are using the button that makes their sprite appear smaller to themselves? basically all xeno queens would use this to avoid clicking themseves and it would be a shame if they were spared from seeing it by using that alternate sprite

**Comment by orthography** [comment]
> will these sprites show up for xeno queens if they are using the button that makes their sprite appear smaller to themselves? basically all xeno queens would use this to avoid clicking themseves and it would be a shame if they were spared from seeing it by using that alternate sprite

Everyone else will still see the xeno queen even if they are in the smaller mode.

**Comment by borb197** [comment]
> > will these sprites show up for xeno queens if they are using the button that makes their sprite appear smaller to themselves? basically all xeno queens would use this to avoid clicking themseves and it would be a shame if they were spared from seeing it by using that alternate sprite
> 
> Everyone else will still see the xeno queen even if they are in the smaller mode.

that was not my question, will the xeno queen see themselves in a maid suit if they are in smaller mode?

File: code/modules/mob/living/carbon/alien/adult/queen.dm [hunk 4]
```diff
@@ -181,3 +181,22 @@
...
+		if(do_after(user, 5 SECONDS, src))
+			maidify()
+			qdel(O)
+		return TRUE
+
+/mob/living/carbon/alien/adult/royal/queen/proc/maidify()
+	name = "alien queen maid"
+	desc = "Lusty, Sexy"
+	icon_state = "alienqmaid"
+	caste = "qmaid"
```
Comment (Militaires) [inline code/modules/mob/living/carbon/alien/adult/queen.dm:200]: it also looks like you took my code from that pr as is, that was my first ever pr and i didnt know how byond works so this var does absolutely nothing

Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:200]: Yeah, 3am posting. <333

The name and sprite works, I need to get the desc to update.

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 1 above)
Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:199]: Sure, I'll send a message.

**Comment by YakumoChen** [comment]
Anyone who sympathises with foul xeno scum deserves their inevitable purging

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 1 above)
Comment (orthography) [reply code/modules/mob/living/carbon/alien/adult/queen.dm]: I did something, it seems to work. Hope it works for you too.

**Comment by orthography** [comment]
> > > > will these sprites show up for xeno queens if they are using the button that makes their sprite appear smaller to themselves? basically all xeno queens would use this to avoid clicking themseves and it would be a shame if they were spared from seeing it by using that alternate sprite
> > > 
> > > 
> > > Everyone else will still see the xeno queen even if they are in the smaller mode.
> > 
> > 
> > that was not my question, will the xeno queen see themselves in a maid suit if they are in smaller mode?
> 
> it doesn't, but if you wanted to it should be easy if OP goes to
> 
> https://github.com/tgstation/tgstation/blob/a4c1f6298e821a6449108e19c2d31c323a7e79cd/code/datums/actions/mobs/small_sprite.dm#L13
> 
> and adds a new action datum except with the small sprite you're describing and either remove the old action and grant that one or you can just have the one action and add a check in its Trigger() for if the icon_state is the maid and change to the correct small sprite if so

Added

File: code/modules/mob/living/carbon/alien/adult/queen.dm [hunk 5]
```diff
@@ -181,3 +184,28 @@
 	if(!silent)
 		to_chat(user, span_noticealien("You discard [src]."))
 	return ..()
+
+/mob/living/carbon/alien/adult/royal/queen/attackby(obj/item/O, mob/user, params)
+	. = ..()
+	if(.)
+		return
+
+	if(istype(O, /obj/item/clothing/under/costume/maid) || istype(O, /obj/item/clothing/under/rank/civilian/janitor/maid))
```
Comment (Militaires) [inline code/modules/mob/living/carbon/alien/adult/queen.dm:193]: you should either have a list of possible uniforms and check against that or have a switch or something if you plan to include more than 1 uniform

**Comment by Militaires** [comment]
last time when i added this it got (rightfully) removed months later for being fetish content so i dont think you'll get this to pass thru but good luck

File: code/modules/mob/living/carbon/alien/adult/queen.dm (hunk 5 above)
Comment (Militaires) [reply code/modules/mob/living/carbon/alien/adult/queen.dm:193]: maybe have a list with each type associated with a costume string like 
list(/obj/item/clothing/under/costume/maid = "maid",  type 2 = "costume 2")
and it sets costume = "whatever string it was associated with" then call dressup() and so on

**Comment by Addust** [comment]
god is dead

**Comment by orthography** [comment]
> last time when i added this it got (rightfully) removed months later for being fetish content so i dont think you'll get this to pass thru but good luck

Addendum.

It's funny.

**Comment by JohnFulpWillard** [comment]
No maintainers really showed interest in merging this (except maybe @LemonInTheDark but he can re-open if he wants), and I don't want to merge it, so I'm gonna close it.

**Comment by OneAsianTortoise** [comment]
Society...

**Comment by CCC23** [comment]
society

**Comment by Burger1998** [comment]
bro this was my most anticipated pr how could you do this

**Pull request closed without merging**